pkg/controller/mysqlcluster/internal/cleaner/pvc_test.go

```diff
@@ -84,66 +82,111 @@ var _ = Describe("Pvc cleaner", func() {
 				MatchLabels: cluster.GetLabels(),
 			},
 		}
+	})
 
-		for i := 0; i < 5; i++ {
-			pvc := corev1.PersistentVolumeClaim{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      fmt.Sprintf("data-%s-mysql-%d", name, i),
-					Namespace: ns,
-					Labels:    cluster.GetLabels(),
-				},
-				Spec: pvcSpec,
-			}
-			pvcs = append(pvcs, pvc)
-		}
+	AfterEach(func() {
+		// remove created cluster
+		c.Delete(context.TODO(), cluster.Unwrap())
+		c.Delete(context.TODO(), secret)
+	})
 
-		for _, pvc := range pvcs {
-			Expect(c.Create(context.TODO(), &pvc)).To(Succeed())
-		}
+	Context("with more PVC than replicas", func() {
+		var (
+			pvcs []corev1.PersistentVolumeClaim
+		)
+		BeforeEach(func() {
 
-		var replicas int32 = 3
-		statefulset = &appsv1.StatefulSet{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      cluster.GetNameForResource(mysqlcluster.StatefulSet),
-				Namespace: cluster.Namespace,
-			},
-			Spec: appsv1.StatefulSetSpec{
-				Replicas: &replicas,
-				Selector: &metav1.LabelSelector{
-					MatchLabels: cluster.GetLabels(),
-				},
-				Template: corev1.PodTemplateSpec{
+			trueVar := true
+			for i := 0; i < 5; i++ {
+				pvc := corev1.PersistentVolumeClaim{
 					ObjectMeta: metav1.ObjectMeta{
-						Labels: cluster.GetLabels(),
+						Name:      fmt.Sprintf("data-%s-mysql-%d", cluster.Name, i),
+						Namespace: cluster.Namespace,
+						Labels:    cluster.GetLabels(),
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: api.SchemeGroupVersion.String(),
+								Kind:       "MysqlCluster",
+								Name:       cluster.Name,
+								UID:        cluster.UID,
+								Controller: &trueVar,
+							},
+						},
 					},
-				},
-				VolumeClaimTemplates: pvcs,
-			},
-		}
+					Spec: pvcSpec,
+				}
+				pvcs = append(pvcs, pvc)
+			}
 
-		Expect(c.Create(context.TODO(), statefulset)).To(Succeed())
+			By("create PVCs")
+			for _, pvc := range pvcs {
+				Expect(c.Create(context.TODO(), &pvc)).To(Succeed())
+			}
+		})
 
-		pvcCleaner = NewPvcCleaner(cluster, options.GetOptions())
-	})
+		AfterEach(func() {
+			for _, pvc := range pvcs {
+				c.Delete(context.TODO(), &pvc)
+			}
+		})
 
-	It("should remove pvcs when cluster is scaled down", func() {
-		Expect(pvcCleaner.Run(context.TODO(), c, nil, rec)).To(Succeed())
-		pvcs2 := &corev1.PersistentVolumeClaimList{}
-		lo := &client.ListOptions{
-			Namespace:     "default",
-			LabelSelector: labels.SelectorFromSet(cluster.GetLabels()),
-		}
-		Expect(c.List(context.TODO(), lo, pvcs2)).To(Succeed())
+		It("should remove extra PVCs when cluster is scaled down", func() {
+			// run cleaner
+			pvcCleaner := NewPVCCleaner(cluster, options.GetOptions(), rec, c)
+			Expect(pvcCleaner.Run(context.TODO())).To(Succeed())
 
-		Expect(3).To(Equal(len(pvcs2.Items)))
+			pvcs2 := listClaimsForCluster(c, cluster)
```

Comment:
```suggestion
			remainingPVCs := listClaimsForCluster(c, cluster)
```
There are multiple occurences of this one.